### ISSUE TITLE
Generate state once per app or once per process

### ISSUE THREAD

**Issue by benjamin-bergia** [okeuday/uuid#18]
Hi,
I am currently using uuid in a webserver to give a UUID to each request that my application receive. Is the `uuid:new/1` a simple seed that I can run once during my application startup and then pass the state around (for example to any child process that would need to generate a UUID) ? Or should I run the `uuid:new/1` inside each new worker process that need to generate a UUID ?

From the code, it looks like `uuid:new/1` is just generating a seed, but the fact that it embeds the process pid make me wonder if I am increasing the change of collisions or loosing any information by running it only once.

**Comment by okeuday** [comment]
@benjamin-bergia You should always call `uuid:new/1` during the initialization of a new Erlang process (e.g., a new socket connection) after the random number generator has been seeded (ideally by calling `quickrand:seed/0`).  To keep latency low, the `mac_address` option can be passed from an app env variable that was set during startup.  An example is at https://github.com/CloudI/cloudi_core/blob/master/src/cloudi_core_i_services_internal.erl#L308-L335 (extra code inbetween) and https://github.com/CloudI/cloudi_core/blob/master/src/cloudi_core_i_app.erl#L102 .

**Comment by benjamin-bergia** [comment]
@okeuday Oh I see, thank you.
I guess the quickrand seed is not needed for UUIDv1, right ?
By the way, it might be worth adding to the doc of `uuid:new/1` (also the part on the quickrand seed).

**Comment by okeuday** [comment]
@benjamin-bergia Sure, I can add something to the README file, to make it clearer.  The quickrand seed is used by the UUIDv1 for the random number used for the `ClockSeq` (in the middle of the UUID v1 data, set at https://github.com/okeuday/uuid/blob/master/src/uuid.erl#L231) and `quickrand:seed/0` helps to avoid common problems that have occurred in the past when seeding the pseudo random number generators (some people have used `os:timestamp/0`, some people have allowed some values to be 0, etc.).

**Comment by benjamin-bergia** [comment]
@okeuday Cool thank you.
I didn't look into quickrand yet but is the `quickrand:seed/0` also a per process operation or a one time thing that I can do in my `application:start/2` ?

**Comment by okeuday** [comment]
@benjamin-bergia `quickrand:seed/0` is a per process operation that can be the line before `uuid:new/1`.  The pseudo random number generators use the process dictionary of each process and `quickrand:seed/0` initializes their seeds.

**Comment by benjamin-bergia** [comment]
@okeuday ok I will add it to my processes then.
Thank you very much for your help.
